### PR TITLE
[fontbe/avar] don't skip no-op segment maps

### LINKED ISSUE #582: GS build with fontc raises OTS error + warning
[fontc source commit](https://github.com/googlefonts/fontc/commit/796d5bdd978ba1877b7f8b738c99ede0603a0a04)
[GS source commit](https://github.com/googlefonts/googlesans/commit/2ca11933d045352af2ada4d4feeacb3e191bb245)

Fonts compiled in the default GH Actions Linux runner environment with the current stable Rust toolchain

```
ERROR: avar: Axis count mismatch
ERROR: Table discarded
WARNING: STAT: Unexpected non-zero offsetToAxisValueOffsets
File sanitized successfully!
```

### PR DESCRIPTION
**Pull request by anthrotype**
GS contains a `GRAD` axis whose avar segment maps does not have any interesting mappings.
It happens to be the last one in the axis order. We were filtering out axis segment maps that contain only identity maps, leading to an avar table with fewer segment maps than there are fvar axes, which is invalid as OTS correctly points out.

So we now still emit these non-interesting segment maps when bulding avar and there's at least one of them that is not an identity map.

Also note that, despite the OT spec allows for empty segment maps for axis that don't distort the default normalization, we want to match fontmake/fonttools and emit the required {-1:1, 0:0, 1:1} mappings even when no other interesting mappings are present for an axis, because apparently some old implementations need that (see https://github.com/fonttools/fonttools/pull/1026).

Fixes #582 avar axis mismatch OTS error

I can confirm that after this, the GS avar table is the same when built with fontc vs fontmake.

### CODE REVIEW

File: fontbe/src/avar.rs [hunk 1]
```diff
@@ -21,7 +22,30 @@ pub fn create_avar_work() -> Box<BeWork> {
...
 }
 
-fn to_segment_map(axis: &Axis) -> Option<SegmentMaps> {
+/// Return true if the SegmentMaps is not a boring identity map
+fn is_interesting(segment_map: &SegmentMaps) -> bool {
+    segment_map
+        .axis_value_maps
+        .iter()
+        .any(|av| av.from_coordinate != av.to_coordinate)
+}
```
Comment (cmyr) [inline fontbe/src/avar.rs:31]: I think this makes sense as a method on the `SegmentMaps` type in write-fonts? I'd probably flip the logic and call it `is_identity`.


One option would be to add the method there, merge this as-is, and open an issue to fix this when we next publish write-fonts.

File: fontbe/src/avar.rs [hunk 2]
```diff
@@ -21,7 +22,30 @@ pub fn create_avar_work() -> Box<BeWork> {
...
+
+/// Return a default avar SegmentMaps containing the required {-1:-1, 0:0, 1:1} maps
+fn default_segment_map() -> SegmentMaps {
```
Comment (cmyr) [inline fontbe/src/avar.rs:34]: and this feels like it should probably be the impl of `Default` for `SegmentMaps`; however we currently autogenerate `Default` impls for all write types, and there's no mechanism to override this, so this is probably fine.

**Review by cmyr — APPROVED** [review]
Looks good, I appreciate all the explanation inline.

File: fontbe/src/avar.rs (hunk 1 above)
Comment (rsheeter) [reply fontbe/src/avar.rs:31]: +1 making the method is_identity

**Comment by rsheeter** [comment]
> I appreciate all the explanation inline

Vigorous +1, it's helpful to us now and incredibly helpful to future editors

**Pull request merged**